deployment/ccip/changeset/v1_6/cs_chain_contracts.go

```diff
@@ -1265,8 +1273,44 @@ func (cfg UpdateRouterRampsConfig) Validate(e deployment.Environment, state chan
 					return err
 				}
 			} else {
-				if err := commoncs.ValidateOwnership(e.GetContext(), cfg.MCMS != nil, e.Chains[chainSel].DeployerKey.From, chainState.Timelock.Address(), chainState.Router); err != nil {
-					return err
+				// If we are activating ramps on the main router, we should validate two things:
+				//   1. All expected CCIP contracts exist on the chain.
+				//   2. All contracts have the expected owner.
+				// That way, if cfg.MCMS exists, we ensure that every contract is owned by MCMS.
+				ownedContracts := map[string]commoncs.Ownable{
```

Comment:
We might need this in other check points. How about creating a method on state which will return a list of expected contracts owned by timelock?